Client/FluentClientExtensions.cs

```diff
@@ -21,8 +21,8 @@ public static class FluentClientExtensions
         public static bool Remove<TFilter>(this List<IHttpFilter> filters)
             where TFilter : IHttpFilter
         {
-            TFilter filter = filters.OfType<TFilter>().FirstOrDefault();
-            return filter != null && filters.Remove(filter);
+            IEnumerable<TFilter> tFilters = filters.OfType<TFilter>();
+            return tFilters.Any() && filters.Remove(tFilters.First());
```

Comment:
LINQ uses deferred execution: `tFilters` doesn't contain the search result, it's an abstract query that's executed *each time* results are requested. You can test that using this code:

```c#
IEnumerable<long> GetTicks()
{
    Thread.Sleep(250);
    yield return DateTime.UtcNow.Ticks;
}

void Main()
{
    IEnumerable<long> ticks = this.GetTicks();
    Console.WriteLine(ticks.First() == ticks.First()); // false
}
```

That means the new version searches the list three times:
* `tFilters.Any()` executes `filters.OfType<TFilter>().Any()`;
* `tFilters.First()` executes `filters.OfType<TFilter>().First()`;
* and `filters.Remove(…)` searches the list for a matching instance.

Forcing immediate execution of the query with `.ToArray()` reduces that to two searches (like the former version):
```c#
IEnumerable<TFilter> tFilters = filters.OfType<TFilter>().Take(1).ToArray();
return tFilters.Any() && filters.Remove(tFilters.First());
```

But since we're okay removing all matches, we can bring that down to one search:
```c#
return filters.RemoveAll(p => p is TFilter) > 0;
```